src/logic/editable-storage-logic.ts

```diff
@@ -18,8 +18,22 @@ export class EditableStorageLogic {
 
     const slots = computeStorageSlots(this.storageLayout, { [variableName]: value });
 
+    // for (const slot of slots) {
+    //   await this.vmManager.putContractStorage(toFancyAddress(this.contractAddress), fromHexString(slot.key), fromHexString(slot.val));
+    // }
+
     for (const slot of slots) {
-      await this.vmManager.putContractStorage(toFancyAddress(this.contractAddress), fromHexString(slot.key), fromHexString(slot.val));
+      let prevStorageValue = await this.vmManager.getContractStorage(toFancyAddress(this.contractAddress), fromHexString(slot.key));
+      let stringValue = remove0x(toHexString(prevStorageValue));
+      if (stringValue && (slot.type === 'address' || slot.type === 'bool' || slot.type.startsWith('contract'))) {
```

Comment:
hmmm interesting code here, is there any way of generalizing this?
i don't think it will cover all use cases, let's say you have an `address, bool, bool` pack, would it work?
did you perform tests on multiple scenarios?

Reply:
only address and bool has same slot value
so if they have slot, previous storage value will be overwritten
so i have done this action